examples/pytorch/language-modeling/run_clm.py

```diff
@@ -413,12 +413,11 @@ def group_texts(examples):
 
     # Training
     if training_args.do_train:
-        if last_checkpoint is not None:
+        checkpoint = None
+        if training_args.resume_from_checkpoint is not None:
+            checkpoint = training_args.resume_from_checkpoint
+        elif last_checkpoint is not None:
```

Comment:
Now probably need to rework this code earlier in the script:
```
    # Detecting last checkpoint.
    last_checkpoint = None
    if os.path.isdir(training_args.output_dir) and training_args.do_train and not training_args.overwrite_output_dir:
        last_checkpoint = get_last_checkpoint(training_args.output_dir)
        if last_checkpoint is None and len(os.listdir(training_args.output_dir)) > 0:
            raise ValueError(
                f"Output directory ({training_args.output_dir}) already exists and is not empty. "
                "Use --overwrite_output_dir to overcome."
            )
        elif last_checkpoint is not None:
            logger.info(
                f"Checkpoint detected, resuming training at {last_checkpoint}. To avoid this behavior, change "
                "the `--output_dir` or add `--overwrite_output_dir` to train from scratch."
            )
```

Otherwise it's now reporting incorrect things. Perhaps putting all of the above into a small function and call it in this context:

```
        if training_args.resume_from_checkpoint is not None:
            checkpoint = training_args.resume_from_checkpoint
        else:
            checkpoint = detect_last_checkpoint()
```

Reply:
No this is all correct. This is the automatic resuming from checkpoint if you pass an `output_dir` that contains a checkpoint, which is the first documented point [here](https://github.com/huggingface/transformers/blob/master/examples/pytorch/README.md#resuming-training).

Reply:
I was addressing the situation for when a user passes an `--output_dir` that contains a checkpoint and `--resume_from_checkpoint` - we aren't preventing this ambiguity and the `logger.info` will report potentially incorrect info, since we are going to ignore `last_checkpoint` that the logger reported will be used.

Reply:
Ah, sorry! I misunderstood your point. Yes the block above should not run when `resume_from_checkpoint` is set.

Reply:
Adapted. Note that the part resulting in the `ValueError` still needs to be run so I only put the test for the logger info.